lib/semantics/type.cc

```diff
@@ -111,7 +111,7 @@ std::ostream &operator<<(std::ostream &o, const ParamValue &x) {
 IntrinsicTypeSpec::IntrinsicTypeSpec(TypeCategory category, int kind)
   : category_{category}, kind_{kind} {
   CHECK(category != TypeCategory::Derived);
-  CHECK(kind > 0);
+  CHECK(kind >= 0);
```

Comment:
Is the significance of kind==0 described somewhere?  Does it mean "default kind for this category"?  If so, the default kinds for each category should be available to you now as part of the semantics context.

Reply:
I've changed it so the kind is always explicit in IntrinsicTypeSpec and in what is written to the .mod file.